scripts/dev_osgeo.sh

```diff
@@ -1,8 +1,8 @@
 #!/bin/bash
 set -e
 
-CRAN=${CRAN_SOURCE:-https://cran.r-project.org}
-echo "options(repos = c(CRAN = '${CRAN}'), download.file.method = 'libcurl')" >>"${R_HOME}/etc/Rprofile.site"
+CRAN=${CRAN_SOURCE:-"https://cloud.r-project.org"}
+echo "options(repos = c(CRAN = '${CRAN}'))" >>"${R_HOME}/etc/Rprofile.site"
```

Comment:
Is it necessary to source install R packages unrelated to proj, gdal, and geos?
I think we can save build time by doing a binary install of as many R packages as possible.
Do you know specifically which R packages only need to be source installed?

Reply:
yeah good question, I was wondering about that as well.  Of course we could one-off install these from source at docker build time without changing the default CRAN repo away from RSPM at all, which would be fine I think.  The difficulty is only if users later upgrade any package binding any of these 3 libraries (which includes some packages not yet on our install list in this script, e.g. `terra`, the successor to `raster`, and will no doubt include other packages in the future).  

I don't know of a good way to ensure R will only use RSPM binaries from packages where the libs come from the official ubuntu repos but source install those packages that have libs built from source (or ppas).  Maybe @jeroen has some advice?  Otherwise I think just turning off RSPM here is safest.

Reply:
Changing the default repository with this script would make sense as you describe.
My concern is that the script seems to install too many R packages (no dependencies with proj, gdal, geos).
Do you think these can be removed from this script?
Or is it possible to install these packages before changing the configuration to do source install?
(I assume most of the packages are already installed in the pre-built image since it runs on `rocker/verse:latest`).

https://github.com/rocker-org/rocker-versioned2/blob/25c29ffb2fc14f21f50a0af02c132ba7bf06015c/scripts/install_dev_osgeo.sh#L125-L151